docs/developer-docs/security/rust-canister-development-security-best-practices.mdx

```diff
@@ -718,12 +718,12 @@ Canisters pay for their cycles which makes them inherently vulnerable to attacks
 
 The correct execution of [canister_inspect_message](../../references/ic-interface-spec#system-api-inspect-message) is not guaranteed in the presence of a malicious node, because it is executed as a [query call](../../references/ic-interface-spec#http-query).
```

Comment:
Would the following make the malicious node issue a bit clearer? 
```suggestion
The correct execution of [canister_inspect_message](../../references/ic-interface-spec#system-api-inspect-message) is not guaranteed because it is executed by a single node and if that node is malicious, it can e.g. simply skip this check. In that case the update call would be executed without any message inspection checks. 
```